p2pserver/block_sync.go

```diff
@@ -745,28 +762,26 @@ func (this *BlockSyncMgr) getFlightBlockCount() int {
 
 func (this *BlockSyncMgr) isBlockOnFlight(blockHash common.Uint256) bool {
 	flightInfos := this.getFlightBlock(blockHash)
-	for _, flightInfo := range flightInfos {
-		if flightInfo != nil {
-			return true
-		}
+	if len(flightInfos) != 0 {
+		return true
 	}
 	return false
 }
 
 //Using polling for load balance
-func (this *BlockSyncMgr) getNextNodeId(nodeList []uint64) uint64 {
+func (this *BlockSyncMgr) getNextNodeId(nextNodeIndex int, nodeList []uint64) (int, uint64) {
 	this.lock.Lock()
 	defer this.lock.Unlock()
```

Comment:
this function has nothing to do with `BlockSyncMgr` now.  remove this lock, and make it a free function